atc/gc/build_log_collector.go

```diff
@@ -23,7 +23,7 @@ func NewBuildLogCollector(
 	batchSize int,
 	buildLogRetentionCalculator BuildLogRetentionCalculator,
 	drainerConfigured bool,
-) Collector {
+) *buildLogCollector {
```

Comment:
nit: Would you mind exporting these types? Having a public constructor return a private type has always felt odd to me. If we expose them it'll be easier to Godoc them in the future.

Reply:
going to hold off on this for now as per our discussion